v1alpha1/proto/grafeas.proto

```diff
@@ -601,6 +601,53 @@ message Discovery {
   Note.Kind analysis_kind = 1;
 }
 
+// In-toto layout
+message Layout {
+  repeated Step steps = 1;
+  google.protobuf.Timestamp expiration_time = 2;
+  string description = 3;
+  repeated Inspect inspection_rules = 4;
+  repeated KeyDetail keys = 5;
+}
+
+// In-toto steps
+message Step {
+  string name = 1;
+  repeated InputOutput expected_materials = 2;
+  repeated InputOutput expected_products = 3;
+  string expected_command = 4;
+  repeated string public_keys = 5;
+  int32 threhold = 6;
+}
+
+message Inspect {
+  string name = 1;
+  repeated InputOutput expected_input = 2;
+  repeated InputOutput expected_output = 3;
+  string run = 4;
+}
+
+// format for materials and products
+message InputOutput {
```

Comment:
I see you are using `InputOutput` for `expected_materials` and `expected_products` in  `Step` and `Inspect` as well as for `materials` and `products` in `IntotoVerification` (i.e. in-toto links). However, these are different things. 

The fields on Steps and Inspections should be lists of artifact rules, parsable by [`artifact_rules.unpack_rule`](https://github.com/in-toto/in-toto/blob/0.2.0/in_toto/artifact_rules.py#L24-L38). These rules are used to specify which artifacts (materials or products) a step should work with, e.g.:
-  *step B's material foo must match (by hash) step A's product foo*, or 
- *step C creates artifact bar*

[An artifact rule example can be found in the demo layout](https://github.com/in-toto/in-toto/blob/0.2.0/tests/demo_files/demo.layout.template#L88-L107).

`materials` and `products` in in-toto links on the other hand, provide the signed evidence of the artifacts a step worked with, used when applying above artifact rules.
[Material and product examples can be found in demo link files](https://github.com/in-toto/in-toto/blob/0.2.0/tests/demo_files/package.2f89b927.link#L22-L31).